lib/backend/etcdv3/etcdv3.go

```diff
@@ -55,13 +57,40 @@ func NewEtcdV3Client(config *apiconfig.EtcdConfig) (api.Client, error) {
 	}
 
 	// Create the etcd client
-	tlsInfo := &transport.TLSInfo{
-		CAFile:   config.EtcdCACertFile,
-		CertFile: config.EtcdCertFile,
-		KeyFile:  config.EtcdKeyFile,
+	// If Etcd Certificate and Key are provided inline through command line agrument,
+	// then the inline values take precedence over the once in the config file.
+	// All the three parametes, Certificate, key and CA certificate are to be provided inline for processing.
+	var tls *tls.Config
+	var err error
+	if (config.EtcdCert != "" && config.EtcdKey == "" && config.EtcdKeyFile != "") ||
```

Comment:
I can't convince myself that this is correct.  Suggest you use this logic:
```
haveInline := config.EtcdCert != "" || config.EtcdKey != "" || config.EtcdCACert != ""
haveFiles := config.EtcdCertFile != "" || config.EtcdKeyFile != "" || config.EtcdCACertFile != ""
if haveInline && haveFiles {
    return nil, fmt.Errorf("cannot mix inline certificate-key and certificate files")
}
```

Reply:
Acknowledged and updated code.